cartography/intel/aws/sagemaker/transform_jobs.py

```diff
@@ -0,0 +1,160 @@
+import logging
+from typing import Any
+from typing import Dict
+from typing import List
+
+import boto3
+import neo4j
+
+from cartography.client.core.tx import load
+from cartography.graph.job import GraphJob
+from cartography.intel.aws.sagemaker.util import extract_bucket_name_from_s3_uri
+from cartography.models.aws.sagemaker.transform_job import (
+    AWSSageMakerTransformJobSchema,
+)
+from cartography.util import aws_handle_regions
+from cartography.util import timeit
+
+logger = logging.getLogger(__name__)
+
+
+@timeit
+@aws_handle_regions
+def get_transform_jobs(
+    boto3_session: boto3.session.Session,
+    region: str,
+) -> List[Dict[str, Any]]:
+    """
+    Get all SageMaker Transform Jobs in the given region.
+    """
+    client = boto3_session.client("sagemaker", region_name=region)
+    paginator = client.get_paginator("list_transform_jobs")
+    transform_jobs: List[Dict[str, Any]] = []
+
+    # Get all transform job names
+    transform_job_names: List[str] = []
+    for page in paginator.paginate():
+        for job in page.get("TransformJobSummaries", []):
+            transform_job_names.append(job["TransformJobName"])
+
+    # Get detailed information for each transform job
+    for job_name in transform_job_names:
+        try:
```

Comment:
I don't think we need to have a `try+except` here because aws_handle_regions takes care of it.